app/models/package.rb

```diff
@@ -318,6 +318,8 @@ def published?
   end
 
   def add_to_stockit
+    return if detail.present? && !detail.valid?
```

Comment:
If there is such a thing as "detail validity" I'd suggest having a validation on the model to prevent it from being persisted in the first place. Or at least raise an error rather than returning silently.

Since this line doesn't throw, it will surely result in future bugs where we ask "why isn't it syncing with Stockit??"

Reply:
agree, removing this code should pass the validation off to Stockit anyway... see the error catching below

Reply:
There is a validity on API side, for this and it doesn't allow to create a record on API. But it is going and creating the record on Stockit. 

That's why I have stopped it here itself.

Reply:
It should still throw rather than return. When writing the function you can't **assume** where and when it's being called, or what else is happening.

Reply:
sure, makes sense.